app.py

```diff
@@ -36,136 +30,195 @@ def send_error_message(self):
 
 app.logger.info('The API is running!')
 
-################# GET-METHODS ##################
+############################################# GET-METHODS ###########################################################
 
+####################### USERS ####################### 
 @app.route('/users', methods=['GET'])
-def users_get():
-    #app.logger.debug(json.dumps(get_users()))
-    return json.dumps(get_users())
+def get_users(full_data=False):
+    if full_data:
+        return json.dumps(users.get_users(full_data=True))
+    else:
+        return json.dumps(users.get_users())
 
 
 @app.route('/users/<user_id>', methods=['GET'])
-def user_get(user_id):
-    #users_list = get_users()
+def get_user(user_id):
+    fields = json.loads(request.data)
     try:
-        user_data = sql_commands.get_user_by_row(rows='*', id_number=user_id)
+        user_data = users.get_user(user_id, fields)
 
         if not user_data:
             raise IdNotFoundError(user_id)
 
-        user = {}
-        user['id'] = user_data[0][0]
-        user['name'] = user_data[0][1]
-        user['last_name'] = user_data[0][2]
-        user['email'] = user_data[0][3]
-        user['date'] = user_data[0][4]
+        return json.dumps(user_data)
 
-        return json.dumps(user)
-        #return json.dumps(search_user_by_id(user_id, users_list))
     except IdNotFoundError as e:
         return json.dumps({'Error': e.send_error_message()}), 404
 
-def get_users(full_data=False):
-    if full_data:
-        users_list = sql_commands.get_users_list(full_data=True)
-    else:
-        users_list = sql_commands.get_users_list()
 
-    return users_list
+##########################################################
 
-def search_user_by_id(id_number, users_list, valid_id=False):
-    for user in users_list:
-        if user['id'] == id_number or user['id'] == int(id_number):
-            if valid_id:
-                return True
-            else:
-                return user
-    
-    raise IdNotFoundError(id_number)
+##################### PRESCRIPTIONS ######################
 
-##########################################################
+@app.route('/prescriptions/<prescription_id>', methods=['GET'])
+def get_prescription(prescription_id):
+    try:
+        prescription = prescriptions.get_prescription(prescription_id)
+
+        if not prescription:
+            raise IdNotFoundError(prescription_id)
+
+        return json.dumps(prescription)
+        
+    except IdNotFoundError as e:
+        return json.dumps({'Error': e.send_error_message()}), 404
 
 
-##################### DELETE-METHODS #######################
+@app.route('/users/<user_id>/prescriptions', methods=['GET'])
+def get_user_prescriptions(user_id):
+    try:
+        if users.get_user(user_id):
+            result = prescriptions.get_prescriptions_by_user(user_id)
+            
+            return json.dumps({'results': result})
```

Comment:
Unify criteria with other endpoints where they all return directly the data, or add this `results` key to the other endpoints